ledgerjs-fio/src/interactions/signTransaction.ts

```diff
@@ -0,0 +1,133 @@
+import type {HexString, ParsedTransaction, ParsedTransferFIOTokensData, Uint8_t, ValidBIP32Path} from "types/internal"
+
+import {InvalidDataReason} from "../errors"
+import type {SignedTransactionData, Version} from "../types/public"
+import {HARDENED, Transaction} from "../types/public"
+import {chunkBy} from "../utils/ioHelpers"
+import {validate} from "../utils/parse"
+import {uint8_to_buf} from "../utils/serialize"
+import {
+    buf_to_hex,
+    date_to_buf,
+    hex_to_buf,
+    path_to_buf,
+    uint16_to_buf,
+    uint32_to_buf,
+    uint64_to_buf,
+} from "../utils/serialize"
+import {INS} from "./common/ins"
+import type {Interaction, SendParams} from "./common/types"
+import {ensureLedgerAppVersionCompatible} from "./getVersion"
+
+const enum P1 {
+    STAGE_INIT = 0x01,
+    STAGE_HEADER = 0x02,
+    STAGE_ACTION_HEADER = 0x03,
+    STAGE_ACTION_AUTHORIZATION = 0x04,
+    STAGE_ACTION_DATA = 0x05,
+    STAGE_WITNESSES = 0x10,
+}
+
+const send = (params: {
+    p1: number,
+    p2: number,
+    data: Buffer,
+    expectedResponseLength?: number
+}): SendParams => ({ins: INS.SIGN_TX, ...params})
+
+
+export function* signTransaction(version: Version, parsedPath: ValidBIP32Path, chainId: HexString, tx: ParsedTransaction): Interaction<SignedTransactionData> {
+    ensureLedgerAppVersionCompatible(version)
+
+    //Initialize and send chainId
+    {
+        const P2_UNUSED = 0x00
+        const response = yield send({
+            p1: P1.STAGE_INIT,
+            p2: P2_UNUSED,
+            data: Buffer.from(chainId, "hex"),
+            expectedResponseLength: 0,
+        })
+    }
+    //Send expiration, ref block num, ref block prefix
+    {
+        const P2_UNUSED = 0x00
+        const response = yield send({
+            p1: P1.STAGE_HEADER,
+            p2: P2_UNUSED,
+            data: Buffer.concat([date_to_buf(tx.expiration), uint16_to_buf(tx.ref_block_num), uint32_to_buf(tx.ref_block_prefix)]),
+            expectedResponseLength: 0,
+        })
+    }
+    //Send action account, name, acount, permission level
+    {
+        const P2_UNUSED = 0x00
+        const contractAccountName: Buffer = Buffer.from(tx.actions[0].contractAccountName, "hex")
+        const response = yield send({
+            p1: P1.STAGE_ACTION_HEADER,
+            p2: P2_UNUSED,
+            data: Buffer.concat([contractAccountName]),
+            expectedResponseLength: 0,
+        })
```

Comment:
Doing `Buffer.concat([buf])` with one buffer is the same as just using that buffer directly.
```suggestion
        yield send({
            p1: P1.STAGE_ACTION_HEADER,
            p2: P2.UNUSED,
            data: Buffer.from(tx.actions[0].contractAccountName, "hex"),
            expectedResponseLength: 0,
        })
```